homeassistant/components/slack/notify.py

```diff
@@ -65,6 +67,20 @@ def _async_sanitize_channel_names(channel_list):
     return [channel.lstrip("#") for channel in channel_list]
 
 
+@callback
+def _async_templatize_blocks(hass, value):
```

Comment:
This function is _very_ similar to `homeassistant.helpers.template.render_complex`: https://github.com/home-assistant/core/blob/5760fb94fe732454df1753fb24e133ca5757745d/homeassistant/helpers/template.py#L66-L74

I didn't use this helper because, although it handles a case similar to Slack's (recursion through a nested structure), it didn't contain any possibility of finding a string and templating it on the fly (what the default case of `async_templatize_blocks` does). Additionally, the presence of the `hass` argument doesn't create a backwards-compatible way to modify `homeassistant.helpers.template.render_complex` (which doesn't accept the HASS argument).

Thoughts on this approach?

Reply:
@MartinHjelmare Could you give me your thoughts here? Apologies for requesting a review when you're so busy, but this should be quick.